### PR TITLE
Run meta data through maybe_unserialize()

### PR DESCRIPTION
**Pull request by ataylorme**
Run meta data through `maybe_unserialize()` as `get_post_custom()` can potentially return serialized data.

See: @link https://codex.wordpress.org/Function_Reference/get_post_custom

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
Do we actually want to unserialize the meta value though? If we're using this for export, wouldn't we prefer strings? Not sure this is an issue. What are you seeing behave wrong?

**Comment by ataylorme** [comment]
Hi @mAAdhaTTah,
There are two parts to this - I probably should have split them.

I've removed the `maybe_unserialize` call. I think a cleaner way to handle that would be to unserialize the meta data and convert it to JSON for storage in GitHub. It's a bit more universal and, in my opinion, easier to work with than serialized data.

The issue I ran into is that `author`, `post_date` and `post_excerpt` from `get_post_custom()` can be serialized and are replacing the initial keys set in the `$meta` array.

Updating the continue check to
```
if ( '_' === substr( $key, 0, 1 ) || isset($meta[$key]) ) {
	continue;
}
```
resolves this by skipping items that are already initially set in the `$meta` array.

Let me know what you think if converting meta data that is an array to JSON and I can submit a separate pull request for that.

File: lib/post.php [hunk 1]
```diff
@@ -331,14 +331,14 @@ public function meta() {
...
+		//convert traditional post_meta values, hide hidden values, skip already populated values
 		foreach ( get_post_custom( $this->id ) as $key => $value ) {
 
-			if ( '_' === substr( $key, 0, 1 ) ) {
+			if ( '_' === substr( $key, 0, 1 ) || isset($meta[$key]) ) {
 				continue;
 			}
 
-			$meta[ $key ] = $value;
+			$meta[$key] = $value;
```
Comment (mAAdhaTTah) [inline lib/post.php]: WordPress coding standards require the space, please don't delete them. Check the Scrutinzr output for other coding style checks.

**Comment by mAAdhaTTah** [comment]
We don't need to convert arrays to JSON, because the YAML frontmatter handles arrays just fine. We can't convert objects into YAML though, because we can't convert YAML into objects on import.

The `isset` check is good, though I haven't run into that before. Can you squash your commits and check [Scrutinizer](https://scrutinizer-ci.com/g/mAAdhaTTah/wordpress-github-sync/inspections/88d3c475-cf31-4143-b0bc-07ddf278802a) for code style fixes?

**Comment by ataylorme** [comment]
@mAAdhaTTah 

> We don't need to convert arrays to JSON, because the YAML frontmatter handles arrays just fine. We can't convert objects into YAML though, because we can't convert YAML into objects on import.

Thank you for the explanation on the YAML front matter.

> The isset check is good, though I haven't run into that before. Can you squash your commits and check Scrutinizer for code style fixes?

Done - commits squashed and Scrutinizer passed.

**Comment by mAAdhaTTah** [comment]
Thanks for your help. This looks good.

**Pull request merged**